### PR TITLE
S2task4

### PR DESCRIPTION
**Pull request by LottieRugg**

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -77,3 +77,20 @@ function selectItem(array $itemsList): string
...
+
+function cleanInput(string $inputData):string {
+    $trimmedInput = trim($inputData);
+    return htmlspecialchars($trimmedInput);
+
+}
+
+
+
+echo cleanInput( "<a href='html'>'specialchars'</a>");
```
Comment (ashleycoles) [inline functions.php]: Don't forget to take this out!

File: data.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+$db = connectToDatabase();
+
+
+if (isset($_POST["name"])) {
+    $name_input = cleanInput($_POST["name"]);
+    $address_input = cleanInput($_POST["address"]);
+    if (($_POST['rating'] >= 0) && ($_POST["rating"] <= 10)) {
+        $rating_input = cleanInput($_POST["rating"]);
+    } else {
+        return "Your rating must be a number from 0 to 10";
```
Comment (ashleycoles) [inline data.php]: This return statement won't actually do anything, it's only possible to return from within a function.

What I'd suggest doing instead is sending another redirect back to the homepage but with some GET data added on containing an error message, kinda like this:

`header("location:pubs_of_sheffield.php?error=Your rating must be a number from 0 to 10");`

You can then check for the presence of an error on the homepage and display it if it's set.

File: data.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<?php
+require 'functions.php';
+$db = connectToDatabase();
+
+
+if (isset($_POST["name"])) {
+    $name_input = cleanInput($_POST["name"]);
```
Comment (ashleycoles) [inline data.php]: Worth doing something similar to your rating check for some of these other fields, for example with $name_input I'd check to make sure it's more than say 3 characters after it's been cleaned (just incase a user enters nothing but a space in the form field)

**Review by ashleycoles — CHANGES_REQUESTED** [review]

File: data.php (hunk 2 above)
Comment (LottieRugg) [reply data.php]: yea i thought the return statement was a bit hopeful tbh

File: data.php [hunk 4]
```diff
@@ -4,17 +4,44 @@
...
+        $local_brewery_input = cleanInput($_POST["local_brewery"]);
+    } else {
+        header("location:pubs_of_sheffield.php?error_brewery=They don't brew beer there! Please enter a valid brewery or leave blank");
+        exit();
+    }
+
     $picture_input = $_POST["picture"];
-    $what_to_order_input = cleanInput($_POST["what_to_order"]);
+
+    if ($_POST["what_to_order"] == "Pint of Carling"){
```
Comment (ashleycoles) [inline data.php]: GLORIOUS!

**Review by ashleycoles — APPROVED** [review]

**Review by ashleycoles — APPROVED** [review]

**Pull request merged**